module1/homework/grayscaleImages/compression.hpp

```diff
@@ -0,0 +1,11 @@
+#pragma once
+#include <array>
+#include <utility>  // for std::pair<>
+#include <vector>
+
+constexpr size_t height = 160;
+constexpr size_t width = 240;
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& imgToCompress);
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(const std::vector<std::pair<uint8_t, uint8_t>>& imgToDecompress);
```

Comment:
```suggestion
using Image = t std::array<std::array<uint8_t, width>, height>;
using compressedImage =  std::vector<std::pair<uint8_t, uint8_t>>;
compressedImage compressGrayscale(const Image& bitmap);
Image decompressGrayscale(const compressedImage& compression);
```

Reply:
Thanks, I'll implement it in STL#2 homework.